thali/thalireplicationmanager.js

```diff
@@ -192,11 +202,44 @@ function muxServerBridge(tcpEndpointServerPort) {
   var serverPlex = multiplex({}, function(stream, id) {
     var clientSocket = net.createConnection({port: tcpEndpointServerPort});
     stream.pipe(clientSocket).pipe(stream);
+
+    clientSocket.on('end', function () {
+      clientSocket.destroy();
+    });
+
+    clientSocket.on('error', function (err) {
+      console.log('incoming client sockiet error %s', err);
+      clientSocket.destroy();
+    });
   });
 
-  return net.createServer(function(incomingClientSocket) {
+  var server = net.createServer(function(incomingClientSocket) {
+    incomingClientSocket.on('end', function () {
+      console.log('incoming client socket end');
```

Comment:
I'm on the assumption that end is perfectly normal but error is not.  Pretty sure we can get rid of that code for destroying the socket because an error should do that already.  However, on stream error we might want to tear down the incoming client socket